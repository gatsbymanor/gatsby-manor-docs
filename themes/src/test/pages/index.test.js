import React from 'react';
import IndexPage from '../../src/pages/index';

describe('<IndexPage />', () => {
  const testData = {
    massivelyImage: 'image',
    lensImage: 'image',
    photonImage: 'image',
    identityImage: 'image',
    dimensionImage: 'image',
    tessellateImage: 'image',
    hero: {
      lead: {
        title: 'Content gallery',
        partOne: 'First content',
        partTwo: 'Content area.',
        callToAction: 'Action',
      },
    },
  };

  const component = shallow(
    <IndexPage  data={testData} />
  );

  it('should render', () => {
    expect(component).toMatchSnapshot();
  });

  it('should have the right props', () => {
    const props = component.instance().props;

    expect(props).toHaveProperty('data');
    expect(props).toHaveProperty('data.hero');
    expect(props).toHaveProperty('data.massivelyImage');
    expect(props).toHaveProperty('data.lensImage');
    expect(props).toHaveProperty('data.photonImage');
    expect(props).toHaveProperty('data.identityImage');
    expect(props).toHaveProperty('data.dimensionImage');
    expect(props).toHaveProperty('data.tessellateImage');  });
});
