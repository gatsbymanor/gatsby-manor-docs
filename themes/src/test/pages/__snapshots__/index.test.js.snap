// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IndexPage /> should render 1`] = `
<div>
   Hello World! Welcome to Docs!
</div>
`;
